pkg/schemadsl/parser/tests/permission_type_annotation.zed

```diff
@@ -0,0 +1,7 @@
+definition mydefinition {
+    relation viewer: user
+    
+    permission view: user = viewer
+    permission edit: user = viewer
+    permission admin: group = viewer
```

Comment:
Add additional tests for error cases:

- `:` but no type before the `=`
- `: something | `
- `: |`